base/essentials.jl

```diff
@@ -222,6 +222,83 @@ macro eval(mod, ex)
     return Expr(:escape, Expr(:call, GlobalRef(Core, :eval), mod, Expr(:quote, ex)))
 end
 
+"""
+    @noscope try ...
+
+Evaluate a `try` block in the local scope.
+
+```jldoctest
+julia> x
+ERROR: UndefVarError: x not defined
+
+julia> @noscope try
+            x = 1
+            error()
+        catch err
+            display(err)
+        finally
+            println("finally")
+        end
+ErrorException()
+finally
+
+julia> x
+1
+```
+"""
+macro noscope(ex)
+    !(isa(ex, Expr) && ex.head == :try) && throw(ArgumentError("@noscope only supports try blocks"))
+    if length(ex.args) == 4 # try that includes finally
+        if ex.args[2] isa Symbol # try-catch-finally with catch var
+            return Expr(:tryfinally,
+                Expr(:trycatch,
+                    :($(esc(ex.args[1]))),
+                    quote
+                        begin
```

Comment:
Actually, I think this should still be scoped
```suggestion
                        let
```